### PR TITLE
feature: authentication via Slack

### PR DESCRIPTION
**Pull request by shockey**
this PR:
* creates a new `POST /auth/access_token` endpoint that grants JWTs
* uses JWT middleware to protect the Channel API
* creates three new env vars: `SLACK_LOGIN_CLIENT_ID`, SLACK_LOGIN_CLIENT_SECRET`, `JWT_SECRET`

---

required by https://github.com/hrxp/frontend/pull/8

### CODE REVIEW

**Comment by shockey** [comment]
[[Feature] "Sign in with Slack"](https://trello.com/c/5gb2fecL/16-feature-sign-in-with-slack)

**Review by spraguejdev — COMMENTED** [review]
Looking good!

File: README.md [hunk 1]
```diff
@@ -5,18 +5,23 @@ API and backend for the HRX Project.
...
-| `PORT`      | The port to expose the API on | No        | `3000`      |
+| Name                        | Description                                                   | Required?              | Default |
+|-----------------------------|---------------------------------------------------------------|------------------------|---------|
+| `MONGO_URL`                 | The MongoDB URL to use                                        | **Yes**                |         |
+| `SLACK_LOGIN_CLIENT_ID`     | Client ID that will be used for "Sign in with Slack" flow     | **Yes**                |         |
+| `SLACK_LOGIN_CLIENT_SECRET` | Client Secret that will be used for "Sign in with Slack" flow | **Yes**                |         |
+| `JWT_SECRET`                | The secret used to sign JWTs                                  | **Yes**, in production |         |
+| `PORT`                      | The port to expose the API on                                 | No                     | `3000`  |
+
+In development, it's most convenient to set these variables in `.env`.
```
Comment (spraguejdev) [inline README.md:22]: Luv it!

**Review by spraguejdev — APPROVED** [review]
Looks good brotha! Thank you for the help.

**Pull request merged**